tests/view_tests/tests/test_i18n_deprecated.py

```diff
@@ -0,0 +1,265 @@
+
+    def test_escaping(self):
+        # Force a language via GET otherwise the gettext functions are a noop!
+        response = self.client.get('/old_jsi18n_admin/?language=de')
+        self.assertContains(response, '\\x04')
+
+    @modify_settings(INSTALLED_APPS={'append': ['view_tests.app5']})
+    def test_non_BMP_char(self):
+        """
+        Non-BMP characters should not break the javascript_catalog (#21725).
+        """
+        with self.settings(LANGUAGE_CODE='en-us'), override('fr'):
+            response = self.client.get('/old_jsi18n/app5/')
+            self.assertContains(response, 'emoji')
+            self.assertContains(response, '\\ud83d\\udca9')
+
+
+@override_settings(ROOT_URLCONF='view_tests.urls')
+@ignore_warnings(category=RemovedInDjango20Warning)
+class JsI18NTestsMultiPackage(SimpleTestCase):
+    """
+    Tests for django views in django/views/i18n.py that need to change
+    settings.LANGUAGE_CODE and merge JS translation from several packages.
+    """
+    @modify_settings(INSTALLED_APPS={'append': ['view_tests.app1', 'view_tests.app2']})
+    def test_i18n_language_english_default(self):
+        """
+        Check if the JavaScript i18n view returns a complete language catalog
+        if the default language is en-us, the selected language has a
+        translation available and a catalog composed by djangojs domain
+        translations of multiple Python packages is requested. See #13388,
+        #3594 and #13514 for more details.
+        """
+        with self.settings(LANGUAGE_CODE='en-us'), override('fr'):
+            response = self.client.get('/old_jsi18n_multi_packages1/')
+            self.assertContains(response, 'il faut traduire cette cha\\u00eene de caract\\u00e8res de app1')
+
+    @modify_settings(INSTALLED_APPS={'append': ['view_tests.app3', 'view_tests.app4']})
+    def test_i18n_different_non_english_languages(self):
+        """
+        Similar to above but with neither default or requested language being
+        English.
+        """
+        with self.settings(LANGUAGE_CODE='fr'), override('es-ar'):
+            response = self.client.get('/old_jsi18n_multi_packages2/')
+            self.assertContains(response, 'este texto de app3 debe ser traducido')
+
+    def test_i18n_with_locale_paths(self):
+        extended_locale_paths = settings.LOCALE_PATHS + [
+            path.join(
+                path.dirname(path.dirname(path.abspath(upath(__file__)))),
+                'app3',
+                'locale',
+            ),
+        ]
+        with self.settings(LANGUAGE_CODE='es-ar', LOCALE_PATHS=extended_locale_paths):
+            with override('es-ar'):
+                response = self.client.get('/old_jsi18n/')
+                self.assertContains(response,
+                    'este texto de app3 debe ser traducido')
+
+
+skip_selenium = not os.environ.get('DJANGO_SELENIUM_TESTS', False)
+
+
+@unittest.skipIf(skip_selenium, 'Selenium tests not requested')
```

Comment:
Please "rebase" to pickup the changes in 44c0ecdd9226d039a8c666b36ae320af2046a1c1.